streaming/src/java/org/apache/hive/streaming/HiveStreamingConnection.java

```diff
@@ -369,21 +429,47 @@ public PartitionInfo createPartitionIfNotExists(final List<String> partitionValu
     String partLocation = null;
     String partName = null;
     boolean exists = false;
+
     try {
       Map<String, String> partSpec = Warehouse.makeSpecFromValues(tableObject.getPartitionKeys(), partitionValues);
       AddPartitionDesc addPartitionDesc = new AddPartitionDesc(database, table, true);
       partName = Warehouse.makePartName(tableObject.getPartitionKeys(), partitionValues);
       partLocation = new Path(tableObject.getDataLocation(), Warehouse.makePartPath(partSpec)).toString();
       addPartitionDesc.addPartition(partSpec, partLocation);
       Partition partition = Hive.convertAddSpecToMetaPartition(tableObject, addPartitionDesc.getPartition(0), conf);
+
+      if (getMSC() == null) {
+        return new PartitionInfo(partName, partLocation, PartitionInfo.Exists.UNKNOWN);
```

Comment:
This can be avoided per previous comment. Add it to set of partitions regardless of whether it exists or not.

Reply:
I see, so what value should `exists` have in this case @prasanthj ?